core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -525,6 +558,7 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
             producerEpochCopy = txnMetadata.producerEpoch
             // PrepareEpochFence has slightly different epoch bumping logic so don't include it here.
             val currentTxnMetadataIsAtLeastTransactionsV2 = !txnMetadata.pendingState.contains(PrepareEpochFence) && txnMetadata.clientTransactionVersion.supportsEpochBump()
+            val incomingEndTxnIsAtLeastTransactionsV2 = !txnMetadata.pendingState.contains(PrepareEpochFence) && clientTransactionVersion.supportsEpochBump()
```

Comment:
Do we need 2 checks?  Can we just check the incoming client version (which we always know here) and use it in all conditions?

Reply:
Updated to only use EndTxn version here.

Reply:
Looks like `endTxnEpochBumped` uses `txnMetadata.clientTransactionVersion.supportsEpochBump()` under the hood.

Reply:
You are saying we should change that one then?

Reply:
Yes, if we use incoming version, we should use it across the board,